source/opt/ir_context.h

```diff
@@ -25,11 +25,24 @@ namespace ir {
 
 class IRContext {
  public:
+  // Available analyses.
+  //
+  // When adding a new analysis:
+  //
+  // 1. Enum values should be powers of 2. These are cast into uint32_t
+  //    bitmasks, so we can have at most 31 analyses represented.
+  //
+  // 2. Make sure it gets invalidated by IRContext methods that add or remove IR
+  //    elements (e.g., KillDef, KillInst, ReplaceAllUsesWith).
+  //
+  // 4. Add handling code in BuildInvalidAnalyses and
+  //    InvalidateAnalysesExceptFor.
   enum Analysis {
-    kAnalysisNone = 0x0,
-    kAnalysisBegin = 0x1,
+    kAnalysisNone = 0 << 0,
+    kAnalysisBegin = 1 << 0,
     kAnalysisDefUse = kAnalysisBegin,
-    kAnalysisEnd = 0x2
+    kAnalysisInstrToBlockMapping = 1 << 1,
+    kAnalysisEnd = kAnalysisInstrToBlockMapping
```

Comment:
I had used kAnalysisEnd as "one-past-the-end".  See the IRContext test.  They do 

 for (Analysis i = IRContext::kAnalysisBegin; i < IRContext::kAnalysisEnd; i <<= 1)

I don't care which way it is, but it something needs to be changed.

Reply:
Ah, yes.  One past the end seems more intuitive.  Fixed.

Also fixed the test AllPreserveFirstOnlyAfterPassWithChange (thanks for the hints!)